### DISCUSSION TITLE
Using NEP-141 as payment

### DISCUSSION THREAD

**Discussion by seaver-choy** [TENK-DAO/tenk#101] [Ideas]
Hi! Would want to ask we can use NEP-141 as payment instead of NEAR token?

**Comment by willemneal** [comment]
Sure!  The issue you need to worry about is the storage cost.  So you would need to either absorb the cost or require attaching something like `0.015 N`.  I also haven't worked out the details myself so there are definitely somethings that require more thought.

For example, you would likely use `ft_transfer_call`, which means implementing [`ft_on_transfer`](https://github.com/near/near-sdk-rs/blob/master/near-contract-standards/src/fungible_token/receiver.rs).
